libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/ranker/BaseDataLayoutCandidateSelector.java

```diff
@@ -0,0 +1,58 @@
+package com.linkedin.openhouse.datalayout.ranker;
+
+import com.linkedin.openhouse.datalayout.strategy.DataLayoutStrategy;
+import java.util.Comparator;
+import java.util.List;
+import java.util.PriorityQueue;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * Base class for data layout candidate selector. It provides the basic functionality to provide
+ * consistent ranking strategy across multiple candidate selection strategies.
+ */
+public abstract class BaseDataLayoutCandidateSelector implements DataLayoutCandidateSelector {
+
+  /**
+   * Rank the candidates to perform data layout optimizations based on the scores.
+   *
+   * @param dataLayoutStrategies all data layout strategies with scores computed.
+   * @return index of the selected data layout strategies ordered by scores.
+   */
+  @Override
+  public List<Integer> select(List<DataLayoutStrategy> dataLayoutStrategies) {
+    PriorityQueue<Pair<DataLayoutStrategy, Integer>> maxHeap =
+        new PriorityQueue<>(
+            new Comparator<Pair<DataLayoutStrategy, Integer>>() {
```

Comment:
Is it significant that element index is stored as well?

Reply:
explained in the comment below.